### PR TITLE
Port VV expansion from Upstream Bee

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
For the love to our downstream from Beestation.
Outdated VV has very serious issue that fools you, giving you incorrect information of some data. This upgrade is, not gonna lie, a must for your sanest life.

Basically port from this: 
* https://github.com/BeeStation/BeeStation-Hornet/pull/10758

For the details, check the original PR
This has been done from Github web VS, so I can't guarantee if it works well. I was bored af at my work.
Also, slightly included minour fixes in addition to the ports (that I originally failed to catch at)

## Changelog
:cl: Various TG coders from multiple PRs (AnturK, Thunder12345, Fikou, Y0SH1M4S73R, Ryll/Shaps, MrStonedOne, LemonInTheDark, ninjanomnom) - port by EvilDragonfiend
fix: Datums and lists used as keys in assoc lists are now linked in vv
add: View variables can now accept input from a colour picker.
code: Changed resolve weakref vv option to TG version
admin: infinity option is added to vv classes
admin: Added the color matrix editor. It is available from a dropdown option in an atom's View Variables menu as well as when editing an existing color matrix.
admin: Added Datum Tagging, a new admin tool for quickly marking and referring back to various datums and atoms. You can tag atoms by ctrl+middle clicking them or using the right click context menu, or from their VV dropdown menu (especially for abstract datums that can't be clicked).
admin: The View-Tags menu can quickly be accessed by F9 default, or using the View-Tags button at the bottom of the Admin -> Game status panel section.
admin: Tagged datums can also be referenced for things like proccalls and VV var edits, similarly to Marked objects.
code: Fixup list helpers, remove listoflist footgun from generic list procs, remove duplicated procs
admin: VV for global vars will now load MUCH faster, in exchange lists are now perma contracted in that particular pane
admin: Appearance vars in VV now display instead of being left blank
fix: Fixes passing in list arguments in callproc
admin: VV can now display the contents of special byond lists like filters, or client.images
admin: VV on images now displays the image in the header
admin: VV can now display filters and includes their type
code: Precalculates special list var names and avoids incorrect vv targets
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
This will work well. Enjoy.
I will appreciate if anyone can help me to show some testing evidences since it's not done from my actual computer.

**Review by Bokkiewokkie — APPROVED** [review]
Very nice stuff, I've enjoyed the added options during the games these last few weeks

File: code/modules/admin/view_variables/debug_variables.dm [hunk 1]
```diff
@@ -1,88 +1,125 @@
...
-			</table></td><td class='rbrak'>&nbsp;</td></tr></tbody></table></span>"} //TODO link to modify_transform wrapper for all matrices
-	else if (istype(value, /datum))
-		var/datum/DV = value
-		if ("[DV]" != "[DV.type]") //if the thing as a name var, lets use it.
-			item = "<a href='?_src_=vars;[HrefToken()];Vars=[REF(value)]'>[VV_HTML_ENCODE(name)] [REF(value)]</a> = [DV] [DV.type]"
-		else
-			item = "<a href='?_src_=vars;[HrefToken()];Vars=[REF(value)]'>[VV_HTML_ENCODE(name)] [REF(value)]</a> = [DV.type]"
+	if(isfilter(value))
+		var/datum/filter_value = value
+		return "/filter (<span class='value'>[filter_value.type] [REF(filter_value)]</span>)"
```
Comment (DeltaFire15) [inline code/modules/admin/view_variables/debug_variables.dm:58]: runtime location

**Review by DeltaFire15 — COMMENTED** [review]
This is causing runtimes on any VV open or refresh, something related to filters apparently.

Hard to miss, open the VV on any object or refresh any VV panel.

File: code/modules/admin/view_variables/debug_variables.dm (hunk 1 above)
Comment (EvilDragonfiend) [reply code/modules/admin/view_variables/debug_variables.dm:58]: Will fix in some hours.

Comment (EvilDragonfiend) [reply code/modules/admin/view_variables/debug_variables.dm:58]: I am not available these days, so sorry, but please live with it for a while. I will bring more up-to-date vv from upstream which is even more awesome, soon™

Comment (Bokkiewokkie) [reply code/modules/admin/view_variables/debug_variables.dm:58]: Ah damn

**Pull request merged**